src/lib/OS/SyscallHandler.cc

```diff
@@ -370,17 +370,30 @@ void SyscallHandler::handleSyscall() {
     }
     case 93: {  // exit
       auto exitCode = currentInfo_.registerArguments[0].get<uint64_t>();
-      std::cout << "[SimEng:SyscallHandler] Received exit syscall: "
-                   "terminating with exit code "
-                << exitCode << std::endl;
-      return concludeSyscall({}, true);
+      uint64_t tid = currentInfo_.threadId;
+      // TODO: When `wait` is supported, return exitCode & 0xFF to parent
+      // TODO: Call all functions registered with `atexit` and `on_exit`
+      // TODO: Flush all open `stdio` streams when supported
+      // TODO: Remove files created by `tmpfile` when supported
+      OS_->terminateThread(tid);
+      std::cerr << "[SimEng:SyscallHandler] Received exit syscall on Thread "
```

Comment:
Think we should continue to print to STDOUT not to STDERR as this isn't an error

Reply:
Done